src/main/java/seedu/address/flashcard/UniqueFlashcardList.java

```diff
@@ -0,0 +1,113 @@
+package seedu.address.flashcard;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.person.exceptions.DuplicatePersonException;
+import seedu.address.model.person.exceptions.PersonNotFoundException;
+
+/**
+ * A list of persons that enforces uniqueness between its elements and does not allow nulls.
+ * A person is considered unique by comparing using {@code Person#isSamePerson(Person)}. As such, adding and updating of
+ * persons uses Person#isSamePerson(Person) for equality so as to ensure that the person being added or updated is
+ * unique in terms of identity in the UniquePersonList. However, the removal of a person uses Person#equals(Object) so
+ * as to ensure that the person with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Flashcard#isSameFlashcard(Flashcard)
+ */
```

Comment:
Could you change the documentation? This is so that others can understand what this class is about.

Reply:
Agree with @joshtyf! We would eventually have to update the documentation anyways, so would be good if we can do it incrementally instead of waiting all the way to the end to update 😄